.github/workflows/ci_reusable_wf.yml

```
@@ -164,21 +164,31 @@ jobs:
             ~/.cargo/git/db
           key: ${{ needs.cargo-registry-cache.outputs.cache-key }}
 
+      - name: Generate build artifacts key
+        id: cache-key
+        shell: bash
+        run: echo "::set-output name=key::$(echo "${{ runner.os }}-${{ steps.rust-toolchain.outputs.rustc }}-${{ github.job }}-${{ needs.cargo-registry-cache.outputs.cache-date }}-${{ hashFiles('**/Cargo.lock') }}")"
+
       - name: Cache build artifacts
         uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed # v2.1.7
         with:
           path: ${{ github.workspace }}/target
-          key: ${{ runner.os }}-${{ steps.rust-toolchain.outputs.rustc }}-${{ github.job }}-${{ needs.cargo-registry-cache.outputs.cache-date }}-${{ hashFiles('**/Cargo.lock') }}
+          key: ${{ steps.cache-key.outputs.key }}
           restore-keys: ${{ runner.os }}-${{ steps.rust-toolchain.outputs.rustc }}-${{ github.job }}-${{ needs.cargo-registry-cache.outputs.cache-date }}-
 
-      - name: cargo clippy
+      - name: Download data
+        run: sh download_data.sh
+
+      - name: Run tests
+        env:
+          RUSTFLAGS: "-D warnings"
         run: |
-          cargo clippy --all-targets -- --deny warnings
-          cargo clippy --all-targets --all-features -- --deny warnings
-          
-  cargo-test:
-    name: cargo-test
-    needs: [cargo-registry-cache, cargo-check]
+          # compilations options must be kept in sync with cargo-tarpaulin and build-linux-lib
+          cargo test --all-features
```

since run consists of only one line now, we could remove the pipe
```suggestion
        # compilations options must be kept in sync with cargo-tarpaulin and build-linux-lib
        run: cargo test --all-features
```